ocs_ci/utility/retry.py

```diff
@@ -5,20 +5,59 @@
 logger = logging.getLogger(__name__)
 
 
+def validate_retry_params(tries, delay, backoff, max_timeout, max_delay):
+    """
+    Validate retry parameters to ensure the total retry time does not exceed max_timeout.
+
+    Args:
+        tries: Number of retry attempts.
+        delay: Initial delay between retries in seconds.
+        backoff: Multiplicative factor for backoff.
+        max_timeout: Maximum total time for retries in seconds.
+        max_delay: Maximum delay between retries in seconds.
+
+    Raises:
+        ValueError: If the retry configuration exceeds the maximum timeout.
+    """
+    total_time = 0
+    current_delay = delay
+
+    for attempt in range(tries):
+        total_time += current_delay
+        if total_time > max_timeout:
+            raise ValueError(
+                f"The retry configuration exceeds the maximum allowed timeout ({max_timeout} seconds) "
+                f"with the given parameters: tries={tries}, delay={delay}, backoff={backoff}, max_delay={max_delay}."
+                f"You can reach up to: {total_time} seconds! Please fix it or adjust parameters of retry."
+            )
+        current_delay = min(current_delay * backoff, max_delay)
+
+
 def retry(
-    exception_to_check, tries=4, delay=3, backoff=2, text_in_exception=None, func=None
+    exception_to_check,
+    tries=4,
+    delay=3,
+    backoff=2,
+    text_in_exception=None,
+    func=None,
+    max_timeout=14400,
```

Comment:
Allowing up to 4 hours of potential delay for one decorator by default still seems like an issue that we should try our best to reduce. Maybe default to 1 hour and explicitly set it to a higher value in use-cases where it's necessary?

Reply:
In this initial PR I didn't want to be that much strict as it will bring me to the situation I will need to touch much more files and potentially breaking something.  So already this high number you see already found out a lot of those bad retries. 

I am also for reducing it more - but for this - we can do in next round - where I will probably just give people list of those by changing values in script and provide output for 1 hour and then let teams to incorporate changes and then reduce time. But I don't want to be stuck with this PR myself for long time and affecting code I am not familiar with.

Reply:
That makes sense to me, I agree with that approach.